Source/Internal/IGListAdapter+UICollectionView.m

```diff
@@ -176,7 +176,12 @@ - (void)collectionView:(UICollectionView *)collectionView didUnhighlightItemAtIn
 
 - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout sizeForItemAtIndexPath:(NSIndexPath *)indexPath {
     IGAssert(![self.collectionViewDelegate respondsToSelector:_cmd], @"IGListAdapter is consuming method also implemented by the collectionViewDelegate: %@", NSStringFromSelector(_cmd));
-    return [self sizeForItemAtIndexPath:indexPath];
+    
+    CGSize size = [self sizeForItemAtIndexPath:indexPath];
+    IGAssert(!isnan(size.height), @"IGListAdapter returned negative height = %f for item at indexPath <%@>", size.height, indexPath);
+    IGAssert(!isnan(size.width), @"IGListAdapter returned negative width = %f for item at indexPath <%@>", size.width, indexPath);
```

Comment:
@yurybogdanov you beat me to it. Would be nice to see a unit test that accounts for this as well in `IGListAdapterTests.m`. Nicely done 👍

Reply:
@vincethecoder thank you for your feedback :)
In fact there is test in the file you mentioned:
`- (void)test_whenSectionControllerReturnsNegativeSize_thatAdapterReturnsZero`

However I will try to add test which covers the case when `NAN` is returned in size. Hope to update PR tomorrow.

Reply:
@yurybogdanov Yep, that's correct. Cool man. 🥇

Reply:
So I spent a couple of days trying to solve this problem with test and it appears that we can't force `sizeForItemAtIndexPath` to return size with NAN (I doubt it can even exist, 'cause I've been trying to `CGSizeMake(NAN, NAN)` and got size with zeros), thus test can't be made.
I can be wrong and if I am any help here would be highly appreciated.

Reply:
you probably want `CGFLOAT_MAX` ?

Reply:
@jessesquires no, it eventually worked with NAN. Just missed one line of code :)
I've updated PR with three tests (when height is NAN and width is positive + vice versa + both width and height are NAN).